frontend/packages/pipelines-plugin/integration-tests/features/pipelines/pipelines-runs.feature

```diff
@@ -255,3 +255,43 @@ Feature: Pipeline Runs
              Then user will see the pipeline run name with failed status
               And user will see failure message below pipeline runs
               And user will also see the log snippet
+
+
+        @regression
+        Scenario Outline: Start pipeline modal with different Workspaces
+            Given pipeline "pipeline-workspace" with PVC "git-PVC" workspace is present
+             When user selects "Start" option from kebab menu for pipeline "pipeline-workspace"
+              And user navigates to Workspaces section
+              And user clicks on dropdown with PVC selected by default to check options
+             Then user sees option Empty Directory, Config Map, Secret, PersistentVolumeClaim, VolumeClaimTemplate
+
+
+        @regression
+        Scenario Outline: Start pipeline modal with Empty directory
+            Given pipeline "pipeline-no-workspace" with no workspace is present
+             When user selects "Start" option from kebab menu for pipeline "pipeline-no-workspace"
+              And user navigates to Workspaces section
+             Then user can see on dropdown with Empty Directory selected by default
+
+
+        @regression
+        Scenario Outline: Show VolumeClaimTemplate options
+            Given pipeline "pipeline-no-workspace" with no workspace is present
```

Comment:
> with no workspace is present

That's not what the rest of this test says. I think you mean no previous runs? Because if it doesn't have a workspace it doesn't have the section / ability to set one.

Reply:
FWIW, the only logic we have today is "Fetch a PVC with the pipeline label" if successful, it selects that. Otherwise it defaults to Empty Directory.

Reply:
Updated